src/Hangfire.Mongo.Tests/MongoDistributedLockFacts.cs

```diff
@@ -112,20 +114,21 @@ public void Ctor_WaitForLock_SignaledAtLockRelease()
         {
             UseConnection(database =>
             {
-                Task.Run(() =>
+                var t = new Thread(() =>
```

Comment:
To be compatible with .netcore on macos ;)
And the real reason is that Task.Run does not necessary run on different threads. And since we are using ThreadLocal<> storage in MongoDistributedLock, the test was failing on macos as the Task.Run would run in the same thread.
The ultimate fix for this is (as @pieceofsummer once pointed out) to use AsyncLocal<>. However that is not supported on .netcore.
So it was down to aither drop support or "tweak" the unit tests.